src/helpers/spaces.ts

```diff
@@ -134,28 +134,40 @@ async function loadSpaces() {
 async function getProposals() {
   const ts = parseInt((Date.now() / 1e3).toFixed());
   const query = `
-    SELECT space, COUNT(id) AS count,
-    COUNT(IF(start < ? AND end > ? AND flagged = 0, 1, NULL)) AS active,
-    count(IF(created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) as count_7d
-    FROM proposals GROUP BY space
+    SELECT
+      space,
+      spaces.proposal_count AS count,
+      COUNT(IF(p.start < ? AND p.end > ? AND p.flagged = 0, 1, NULL)) AS active,
+      COUNT(IF(p.created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) AS count_7d,
+    FROM proposals p
+    JOIN spaces ON spaces.id = space
+    GROUP BY space
   `;
   return await db.queryAsync(query, [ts, ts]);
 }
 
 async function getVotes() {
   const query = `
-    SELECT space, COUNT(id) as count,
-    count(IF(created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) as count_7d
-    FROM votes GROUP BY space
+    SELECT
+      space,
+      spaces.vote_count as count,
+      COUNT(IF(v.created > (UNIX_TIMESTAMP() - 604800), 1, NULL)) AS count_7d
+    FROM votes v
+    JOIN spaces ON spaces.id = space
+    GROUP BY space
```

Comment:
This one is taking more than 20 mins (still running) but the previous one took around 12.5 mins

Reply:
It took 24 mins

Reply:
How can a simple read on a column `vote_count` be slower than a `COUNT(id)` ?

Reply:
Maybe the slowness is not related to the update, and the whole query itself does not take same time on each run ? (sometime 12min, sometime 20min, sometime 30min) ?

Reply:
I think slowness could be because of JOIN spaces ON spaces.id = space, adding spaces to all votes

Reply:
we already have all space data in `mapSpaces` so we don't need to get this data here, we can just get `7d` counts here and we should be good

Reply:
Yes, i found a better query, everything will be done in less than 10s